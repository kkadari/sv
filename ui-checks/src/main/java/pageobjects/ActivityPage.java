package pageobjects;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.How;

import java.util.List;

public class ActivityPage extends PageObject {

    @FindBy(how = How.CSS, using = ".j-act-title")
    List<WebElement> divTitles;

    public ActivityPage(WebDriver driver){
        super(driver);
    }

    public String returnStreamListHtml(){
        String streams = "";

        for(WebElement title : divTitles){
            streams += title.getAttribute("outerHTML");
        }

        return streams;
    }

}
