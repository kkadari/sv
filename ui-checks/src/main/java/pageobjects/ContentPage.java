package pageobjects;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.How;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.util.List;

public class ContentPage extends PageObject {

    @FindBy(how = How.CSS, using = ".j-td-title")
    List<WebElement> tableCellTitle;

    public ContentPage(WebDriver driver){
        super(driver);
    }

    public String returnTableRowContentHtml(){
        WebDriverWait wait = new WebDriverWait(driver, 10);
        wait.until(ExpectedConditions.visibilityOf(tableCellTitle.get(0)));

        String tableRowToReturn = "";

        for(WebElement row : tableCellTitle){
            tableRowToReturn += row.getAttribute("outerHTML");
        }

        return tableRowToReturn;
    }

}
