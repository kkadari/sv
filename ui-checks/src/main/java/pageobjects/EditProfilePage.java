package pageobjects;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.How;

public class EditProfilePage extends PageObject {

    @FindBy(how = How.CSS, using = "textarea")
    WebElement txtAreaProfessionalProfile;

    @FindBy(how = How.CSS, using = "input[name='save']")
    WebElement buttonSave;

    public EditProfilePage(WebDriver driver){
        super(driver);
    }

    public void setTxtAreaProfessionalProfile(String text){
        txtAreaProfessionalProfile.sendKeys(text);
    }

    public ProfilePage clickButtonSave(){
        buttonSave.click();

        return new ProfilePage(driver);
    }
}
