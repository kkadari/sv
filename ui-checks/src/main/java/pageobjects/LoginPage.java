package pageobjects;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.How;

public class LoginPage extends PageObject {

    @FindBy(how = How.CSS, using = "#username01")
    WebElement txtUsername;

    @FindBy(how = How.CSS, using = "#password01")
    WebElement txtPassword;

    @FindBy(how = How.CSS, using = "#login-submit")
    WebElement inptSubmit;

    public LoginPage(WebDriver driver){
        super(driver);
    }

    public void setUsername(String username){
        txtUsername.sendKeys(username);
    }

    public void setPassword(String password){
        txtPassword.sendKeys(password);
    }

    public void clickSubmit(){
        inptSubmit.click();
    }

}
