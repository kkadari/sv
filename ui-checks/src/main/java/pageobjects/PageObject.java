package pageobjects;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.support.PageFactory;
import org.openqa.selenium.support.ui.WebDriverWait;

public class PageObject {

    protected WebDriver driver;
    protected int timeout = 10;

    public PageObject(WebDriver driver)
    {
        this.driver = driver;
        PageFactory.initElements(this.driver, this);
    }

}
