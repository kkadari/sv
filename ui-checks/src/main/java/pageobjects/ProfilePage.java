package pageobjects;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.How;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class ProfilePage extends PageObject {

    @FindBy(how = How.CSS, using = ".profile-data")
    WebElement sectionProfileData;

    public ProfilePage(WebDriver driver){
        super(driver);
        WebDriverWait wait = new WebDriverWait(driver, 10);
        wait.until(ExpectedConditions.visibilityOf(sectionProfileData));
    }
}
