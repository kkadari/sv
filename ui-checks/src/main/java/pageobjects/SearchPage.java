package pageobjects;

import org.openqa.selenium.NoSuchElementException;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.How;

public class SearchPage extends PageObject {

    @FindBy(how = How.CSS, using = ".j-content-results")
    WebElement divMainResults;

    @FindBy(how = How.CSS, using = ".j-search-result-summary")
    WebElement divResultContent;

    public SearchPage(WebDriver driver){
        super(driver);
    }

    public String returnSearchResults(){
        return divMainResults.getAttribute("outerHTML");
    }

    public boolean returnResultIsDisplayed(){
        try {
            return divResultContent.isDisplayed();
        } catch (NoSuchElementException e){
            return false;
        }
    }

}
