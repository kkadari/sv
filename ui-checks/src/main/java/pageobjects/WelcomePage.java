package pageobjects;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.How;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class WelcomePage extends PageObject {

    @FindBy(how = How.CSS, using = ".content-small .widget-container")
    WebElement divWidgetContainer;

    @FindBy(how = How.CSS, using = "#whats-happening")
    WebElement divWhatsHappening;

    @FindBy(how = How.CSS, using = "#feed-popular")
    WebElement listFeedPopular;

    @FindBy(how = How.CSS, using = "#feed-following")
    WebElement listFeedFollowing;

    @FindBy(how = How.CSS, using = ".event-content")
    WebElement divContent;

    @FindBy(how = How.CSS, using = ".footer-main")
    WebElement divFooter;

    @FindBy(how = How.CSS, using = "#customBrowseMenuAnchor")
    WebElement linkContentMenu;

    @FindBy(how = How.CSS, using = "#customBrowseMenu")
    WebElement divBrowseMenu;

    @FindBy(how = How.CSS, using = "#customBrowseMenu .j-pop-desc")
    WebElement divContentMenuItem;

    @FindBy(how = How.CSS, using = "#customMembersMenuAnchor")
    WebElement linkMembersMenu;

    @FindBy(how = How.CSS, using = "#customMembersMenu")
    WebElement divMembersMenu;

    @FindBy(how = How.CSS, using = "#navCommunitiesAndNodes")
    WebElement linkCommunityMenu;

    @FindBy(how = How.CSS, using = "#menuCommunitiesAndNodes")
    WebElement divCommunityMenu;

    @FindBy(how = How.CSS, using = "#menuCommunitiesAndNodes .j-pop-desc")
    WebElement divCommunityMenuItem;

    private WebDriverWait wait = new WebDriverWait(driver, timeout);

    public WelcomePage(WebDriver driver){
        super(driver);
    }

    public String returnWidgetContainerHtml(){
        return divWidgetContainer.getAttribute("outerHTML");
    }

    public String returnWhatsHappeningHtml(){
        wait.until(ExpectedConditions.visibilityOf(divContent));

        return divWhatsHappening.getAttribute("outerHTML");
    }

    public void clickFeedPopular() {
        wait.until(ExpectedConditions.visibilityOf(listFeedPopular));

        listFeedPopular.click();
    }

    public void clickFeedFollowing() {
        wait.until(ExpectedConditions.visibilityOf(listFeedFollowing));

        listFeedFollowing.click();
    }

    public String returnFooterHtml() {
        return divFooter.getAttribute("outerHTML");
    }

    public void clickContent() {
        linkContentMenu.click();
    }

    public String returnContentMenuPopupHtml(){
        wait.until(ExpectedConditions.visibilityOf(divContentMenuItem));

        return divBrowseMenu.getAttribute("outerHTML");
    }

    public void clickMembers() {
        linkMembersMenu.click();
    }

    public String returnMembersMenuPopupHtml() {
        return divMembersMenu.getAttribute("outerHTML");
    }

    public void clickCommunityAndNodes() {
        linkCommunityMenu.click();
    }

    public String returnCommunityAndNodesPopupHtml() {
        wait.until(ExpectedConditions.visibilityOf(divCommunityMenuItem));

        return divCommunityMenu.getAttribute("outerHTML");
    }

}
