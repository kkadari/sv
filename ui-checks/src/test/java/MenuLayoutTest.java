import org.approvaltests.Approvals;
import org.junit.Test;
import pageobjects.WelcomePage;

public class MenuLayoutTest extends TestSetup{

    @Test
    public void approveCustomFooter(){
        driver.navigate().to(baseUrl + "/welcome");

        WelcomePage welcome = new WelcomePage(driver);
        String footerHtml = welcome.returnFooterHtml();

        Approvals.verifyHtml(footerHtml);
    }

    @Test
    public void approveContentMenu(){
        driver.navigate().to(baseUrl + "/welcome");

        WelcomePage welcome = new WelcomePage(driver);
        welcome.clickContent();
        String menuHtml = welcome.returnContentMenuPopupHtml();

        Approvals.verifyHtml(menuHtml);
    }

    @Test
    public void approveMembersMenu(){
        driver.navigate().to(baseUrl + "/welcome");

        WelcomePage welcome = new WelcomePage(driver);
        welcome.clickMembers();
        String menuHtml = welcome.returnMembersMenuPopupHtml();

        Approvals.verifyHtml(menuHtml);
    }

    @Test
    public void approvePlacesMenu(){
        driver.navigate().to(baseUrl + "/community/support/user-feedback");
        driver.navigate().to(baseUrl + "/community/support/user-guidance");
        driver.navigate().to(baseUrl + "/community/faq");
        driver.navigate().to(baseUrl + "/community/support");

        driver.navigate().to(baseUrl + "/welcome");

        WelcomePage welcome = new WelcomePage(driver);
        welcome.clickCommunityAndNodes();
        String menuHtml = welcome.returnCommunityAndNodesPopupHtml();

        Approvals.verifyHtml(menuHtml);
    }
}
