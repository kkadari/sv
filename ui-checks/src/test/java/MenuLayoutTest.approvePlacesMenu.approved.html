<div id="menuCommunitiesAndNodes" class="j-pop-main j-menu j-quick-menu" style="display: block;"><span class="j-508-label a11y-boundary">Menu begins</span><div class="j-pop-desc"><h2>Places</h2></div><div><span class="j-508-label a11y-boundary">Menu begins</span><ul id="myPlaces-list"><li><a href="/places" class="custom-dropdown j-ui-elem active" title="All places">All</a></li><li><a href="/places?filterID=following" class="custom-dropdown j-ui-elem active" title="Places you follow">Following</a></li><li><a href="/places?filterID=member" class="custom-dropdown j-ui-elem active" title="places you are a member of">Member</a></li><li><a href="/places?filterID=owner" class="custom-dropdown j-ui-elem active" title="Places you are an owner of">Owner</a></li></ul><span class="j-508-label a11y-boundary">Menu ends</span></div>	<div><strong class="custom-dropdown-heading font-color-meta-light">Recently Viewed</strong><span class="j-508-label a11y-boundary">Menu begins</span><ul id="myPlaces-list"><li><a class="custom-dropdown j-ui-elem active" href="/community/support" title="Support">Support</a></li><li><a class="custom-dropdown j-ui-elem active" href="/community/support/user-guidance" title="User Guidance">User Guidance</a></li><li><a class="custom-dropdown j-ui-elem active" href="/community/support/user-feedback" title="User Feedback">User Feedback</a></li><li><a class="custom-dropdown j-ui-elem active" href="/community/support/faq" title="FAQ">FAQ</a></li></ul><a class="custom-dropdown j-ui-elem active show-more" href="/places?filterID=historyarchetype[containers]" title="More places">view more</a><span class="j-508-label a11y-boundary">Menu ends</span></div><span class="j-508-label a11y-boundary">Menu begins</span></div>
